00xx-account-deletion.md

```diff
@@ -0,0 +1,576 @@
+
+This simple change trickles down into updating the transitive closure of all
+type definitions depending on `Account_update`, including
+`Mina_wire_types.Mina_transaction`.
+
+#### Effect of deleting an account
+
+Deleting an account $a₀$ results in two actions:
+1. Actual removal of account $a₀$ of the storage layer;
+2. Return of balance of MINA tokens from $a₉$ and initial creation fee to a specified $a₁$ account.
+
+Handling other tokens is the responsability of the smart contract(s) dealing with these tokens.
+
+At the snark level, deletion should thus be equivalent to proving two things
+1. The location $l$ of account $a₀$ is now  equal to the empty account.
+   A simple implementation is to call [`set_account`](https://github.com/MinaProtocol/mina/blob/4495af5caea5e1bb2f98f92592c065f93a586ade/src/lib/transaction_snark/transaction_snark.ml#L1531) to location $l$ with the empty account;
+2. The change in MINA of $a₁$ should be equal to
+   - the initial creation fee;
+   - augmented by any remaining MINA balance.
+
+
+**Permissions**. Furthermore, account deletion should not be permitted except
+for some authorized participants, e.g., the contract that created the consumable
+account. We will extend the [permissions
+type](https://github.com/MinaProtocol/mina/blob/4495af5caea5e1bb2f98f92592c065f93a586ade/src/lib/mina_base/permissions.ml#L343)
+with a dedicated `delete` field.  The handling of permissions will be similar as
+the one for [nonce
+update](https://github.com/MinaProtocol/mina/blob/4495af5caea5e1bb2f98f92592c065f93a586ade/src/lib/transaction_logic/zkapp_command_logic.ml#L1687)
+for the computation of `has_permission` and the updated `local_state`. This
+updated local state will also depend on extending [`Failure.t`](https://github.com/MinaProtocol/mina/blob/4495af5caea5e1bb2f98f92592c065f93a586ade/src/lib/mina_base/transaction_status.ml#L9) with an extra variant `Update_not_permitted_delete`.
+
+
+
+
+<!-- user command snark https://github.com/MinaProtocol/mina/blob/develop/src/lib/transaction_logic/mina_transaction_logic.ml#L1007
+ !-- zkapp command snark https://github.com/MinaProtocol/mina/blob/develop/src/lib/transaction_logic/zkapp_command_logic.ml#L1232 -->
+
+### Updating the archive node
+
+The addition of a field in `Account_update` will need to reflected in table
+`zkapp_account_update_body` with the addition of a `delete_account` column.
+
+The migration of the current database needs to set a `false` value in this new
+column for all already registered rows.
+
+
+
+### Other expected code changes
+#### Ledgers
+
+#### Removal
+
+We propose to support removing elements in ledgers through  2 functions, that we specify below:
+
+- `val remove_location: t -> location -> unit`: remove the account found at the
+  given `location` from the ledger, don't do anything if `location` is occupied
+  by the empty account
+- `val remove_account: t -> account -> unit`: remove `account` from the ledger,
```

Comment:
What is `account` here? `account_id` perhaps? Or do you pass a full account that might be in the ledger?

Reply:
It's an abstract `account` type as defined [here](C). 

Come to think of it an `account_id`might be better.

Reply:
Updated